Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/NodeGraph.cpp

```diff
@@ -2397,14 +2403,76 @@ namespace EMStudio
         FitGraphOnScreen(graphWidgetRect.width(), graphWidgetRect.height(), QPoint(0, 0), false);
     }
 
-    void NodeGraph::RenderNodeGroups(QPainter& painter)
+    QRect NodeGraph::ComputeNodeGroupRect(const EMotionFX::AnimGraphNodeGroup* nodeGroup) const
+    {
+        EMotionFX::AnimGraphNode* currentNode =
+            m_currentModelIndex.data(AnimGraphModel::ROLE_NODE_POINTER).value<EMotionFX::AnimGraphNode*>();
+
+        QRect groupRect;
+        QRect nodeRect;
+
+        int32 top = std::numeric_limits<int32>::max();
+        int32 bottom = std::numeric_limits<int32>::lowest();
+        int32 left = std::numeric_limits<int32>::max();
+        int32 right = std::numeric_limits<int32>::lowest();
+
+        // get the number of nodes inside the node group and skip the group in case there are no nodes in
+        const size_t numNodes = nodeGroup->GetNumNodes();
+        if (numNodes == 0)
+        {
+            return groupRect;
+        }
+
+        bool nodesInGroupDisplayed = false;
+        for (size_t j = 0; j < numNodes; ++j)
+        {
+            // get the graph node by the id and skip it if the node is not inside the currently visible node graph
+            const EMotionFX::AnimGraphNodeId nodeId = nodeGroup->GetNode(j);
+            EMotionFX::AnimGraphNode* node = currentNode->RecursiveFindNodeById(nodeId);
+            GraphNode* graphNode = FindGraphNode(node);
+            if (graphNode)
```

Comment:
I find using the assignment operator `=` in conditional expressions confusing because it's easy to misread as `==`. I think the risk of this particular variable getting misused outside of the `if` block is small in this case.